config/initializers/backtrace_silencers.rb

```diff
@@ -2,7 +2,8 @@
 # Be sure to restart your server when you modify this file.
 
 # You can add backtrace silencers for libraries that you're using but don't wish to see in your backtraces.
-# Rails.backtrace_cleaner.add_silencer { |line| line =~ /my_noisy_library/ }
+# Rails.backtrace_cleaner.add_silencer { |line| /my_noisy_library/.match?(line) }
 
-# You can also remove all the silencers if you're trying to debug a problem that might stem from framework code.
-# Rails.backtrace_cleaner.remove_silencers!
+# You can also remove all the silencers if you're trying to debug a problem that might stem from framework code
+# by setting BACKTRACE=1 before calling your invocation, like "BACKTRACE=1 ./bin/rails runner 'MyClass.perform'".
+Rails.backtrace_cleaner.remove_silencers! if ENV['BACKTRACE']
```

Comment:
Do we use `BACKTRACE` the env variable? Is this required for rails 7 upgrade?

Reply:
inspired by the following
https://github.com/rails/rails/pull/50563

this has been merged, but it will be made effective starting Rails 7.2..
This environment variable is useful if we want to debug the framework code easier